src/app/data-model-provider/Provider.h

```diff
@@ -105,6 +105,16 @@ class Provider : public ProviderMetadataTree
     virtual std::optional<ActionReturnStatus> Invoke(const InvokeRequest & request, chip::TLV::TLVReader & input_arguments,
                                                      CommandHandler * handler) = 0;
 
+    /// Report attribute changed
+    ///
+    /// When this is invoked, the caller is expected to increase the cluster data version and the attribute path
+    /// should be marked as dirty by the data model provider listener so that the reporter can report the attribute
+    /// changes to the subscriber.
+    ///
+    /// It should be invoked when attribute managed by access override is changed but the change is not caused by
+    /// WriteAttribute(Such as LastNetworkingStatus in NetworkCommissioning cluster).
+    virtual void ReportAttributeChanged(const AttributePathParams & path) = 0;
```

Comment:
This feels off: the data model provider is supposed to be the "owner of the data" including all implementations of WriteAttribute. The fact that it uses AttributeAccessInterface for example should be transparent. 

I do not think this is the right placement for a method as it seems to invert dependencies at a first glance.

Reply:
```suggestion
   [[deprecated("Workaround until all code can use a context object instead")]]
    virtual void Temporary_ReportAttributeChanged(const AttributePathParams & path) = 0;
```

With some documentation explaining this will be replaced.

Reply:
Added the documentation